packages/frontend-plugin-api/src/components/ComponentsContext.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentRef } from '@backstage/frontend-plugin-api';
+import {
+  createVersionedContext,
+  createVersionedValueMap,
+  useVersionedContext,
+} from '@backstage/version-bridge';
+import React from 'react';
+import { ComponentType, PropsWithChildren } from 'react';
+
+/** @public */
+export type ComponentsContextValue = Record<string, ComponentType<any>>;
+
+const ComponentsContext = createVersionedContext<{ 1: ComponentsContextValue }>(
```

Comment:
I've actually been thinking that we should be moving away from context and use Utility APIs for more of these kind of things instead. What do you think about making this an API instead?

Reply:
Converted to components API